src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp

```diff
@@ -267,65 +223,59 @@ void TypeLookupManager::remove_remote_endpoints(
     }
 }
 
-StatefulWriter* TypeLookupManager::get_builtin_request_writer()
+fastrtps::rtps::SampleIdentity TypeLookupManager::get_type_dependencies(
+        const fastrtps::types::TypeIdentifierSeq& id_seq) const
 {
-    return builtin_request_writer_;
-}
+    fastrtps::rtps::SampleIdentity id = INVALID_SAMPLE_IDENTITY;
 
-StatefulWriter* TypeLookupManager::get_builtin_reply_writer()
-{
-    return builtin_reply_writer_;
-}
+    TypeLookup_getTypeDependencies_In in;
+    //TODO with xtypes::TypeIdentifierSeq
+    // in.type_ids() = id_seq;
+    (void) id_seq;
+    TypeLookup_RequestPubSubType type;
+    TypeLookup_Request* request = static_cast<TypeLookup_Request*>(type.createData());
+    request->data().getTypeDependencies(in);
 
-WriterHistory* TypeLookupManager::get_builtin_request_writer_history()
-{
-    return builtin_request_writer_history_;
-}
-
-WriterHistory* TypeLookupManager::get_builtin_reply_writer_history()
-{
-    return builtin_reply_writer_history_;
+    if (send_request(*request))
+    {
+        id = get_rtps_sample_identity(request->header().requestId());
+    }
+    type.deleteData(request);
+    return id;
 }
 
-StatefulReader* TypeLookupManager::get_builtin_request_reader()
+fastrtps::rtps::SampleIdentity TypeLookupManager::get_types(
+        const fastrtps::types::TypeIdentifierSeq& id_seq) const
 {
-    return builtin_request_reader_;
-}
+    fastrtps::rtps::SampleIdentity id = INVALID_SAMPLE_IDENTITY;
 
-StatefulReader* TypeLookupManager::get_builtin_reply_reader()
-{
-    return builtin_reply_reader_;
-}
+    TypeLookup_getTypes_In in;
+    //TODO with xtypes::TypeIdentifierSeq
+    // in.type_ids() = id_seq;
+    (void) id_seq;
+    TypeLookup_RequestPubSubType type;
+    TypeLookup_Request* request = static_cast<TypeLookup_Request*>(type.createData());
+    request->data().getTypes(in);
 
-ReaderHistory* TypeLookupManager::get_builtin_request_reader_history()
-{
-    return builtin_request_reader_history_;
+    if (send_request(*request))
+    {
+        id = get_rtps_sample_identity(request->header().requestId());
+    }
+    type.deleteData(request);
+    return id;
 }
 
-ReaderHistory* TypeLookupManager::get_builtin_reply_reader_history()
+ReturnCode_t TypeLookupManager::async_get_type(
+        xtypes::TypeInformation,
+        GuidPrefix_t,
+        AsyncGetTypeCallback&)
 {
-    return builtin_reply_reader_history_;
+    return RETCODE_OK;
```

Comment:
Return `RETCODE_UNSUPPORTED` until the method is implemented.

Reply:
Fixed in [7dafd25](https://github.com/eProsima/Fast-DDS/pull/4126/commits/7dafd25b2d97a27f2f53acf522b0d2c68e72a46e)